students/uss_tomasz/lesson_12_debugging/12_3_valdated_user_base/test_validate_user_base.py

```diff
@@ -0,0 +1,133 @@
+import os
+import shutil
+import tempfile
+from validate_user_base import parse_args, is_record_correct, validate_registration, csv_existence_checker, \
+    get_csv_content, is_registration_inside_csv, csv_entry, execute_script
+
+BASE_NAME = 'temporary_database.csv'
+
+
+def test_parse_args():
+    # Test default:
+    assert parse_args(['-e', 'mail@gmail.com', '-p', 'pass']) == {
+        'email': 'mail@gmail.com', 'password': 'pass', 'telephone': '000-000-000', 'zip_code': '00-000'}
+    # Test various inputs:
+    assert parse_args(['-e', 'mail@gmail.com', '-p', 'pass', '-t', '123-456-789', '-z', '90-210']) == {
+        'email': 'mail@gmail.com', 'password': 'pass', 'telephone': '123-456-789', 'zip_code': '90-210'}
+    # Test missing email:
+    assert parse_args(['-p', 'OnlyPassword1']) == {}
+    # Test missing password:
+    assert parse_args(['-e', 'mail@gmail.com']) == {}
+
+
+def test_is_record_correct():
+    assert is_record_correct('email', '$@gmail.com') is False
+    assert is_record_correct('email', 'user@$.com') is False
+    assert is_record_correct('email', 'user@gmail.com') is True
+
+    assert is_record_correct('password', 'Admin11') is False  # too short
```

Comment:
if you pass error cause as string parameter
```
assert not is_record_correct('password', 'Admin11'), 'too short'
```
You'll get error message in assertion error. It's much more convenient than maintaining comments
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AssertionError: too short
```